src/scribe_data/checkquery.py

```diff
@@ -0,0 +1,442 @@
+
+    return queries
+
+
+def changed_queries() -> Optional[list[QueryFile]]:
+    """Find all the SPARQL queries that have changed.
+
+    Includes new queries.
+
+    Returns:
+        Optional[list[QueryFile]]: list of changed/new SPARQL queries or None if error.
+    """
+
+    result = subprocess.run(
+        (
+            "git",
+            "status",
+            "--short",
+        ),
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        universal_newlines=True,
+        check=False,
+    )
+
+    if result.returncode != EXIT_SUCCESS:
+        print(f"ERROR: {result.stderr}", file=sys.stderr)
+        return None
+
+    # example 'result.stdout'
+    #
+    # M a/b/c/changed.sparql
+    # ?? ../new.sparql
+
+    changed_files = [
+        Path(norm_line.split(maxsplit=1)[1]).resolve()
+        for line in result.stdout.split("\n")
+        if (norm_line := line.strip())
+    ]
+
+    return [QueryFile(fpath) for fpath in changed_files if fpath.suffix == ".sparql"]
+
+
+def sparql_context(url: str) -> SPARQL.SPARQLWrapper:
+    """Configure a SPARQL context.
+
+    A context allows the execution of SPARQL queries.
+
+    Args:
+        url (str): a valid URL of a SPARQL endpoint.
+
+    Returns:
+        SPARQLWrapper: the context.
+    """
+    context = SPARQL.SPARQLWrapper(url)
+    context.setReturnFormat(SPARQL.JSON)
+    context.setMethod(SPARQL.POST)
+
+    return context
+
+
+def execute(
+    query: QueryFile, limit: int, context: SPARQL.SPARQLWrapper, tries: int = 3
+) -> dict:
+    """Execute a SPARQL query in a given context.
+
+    Args:
+        query (QueryFile): the SPARQL query to run.
+        limit (int): the maximum number of results a query should return.
+        context (SPARQLWrapper): the SPARQL context.
+        tries (int): the maximum number of times the query should be executed
+                    after failure.
+
+    Returns:
+        dict: results of the query.
+    """
+
+    def delay_in_seconds() -> int:
+        """How long to wait, in seconds, between executing repeat queries."""
+        return int(math.ceil(10.0 / math.sqrt(tries)))
+
+    if tries <= 0:
+        raise QueryExecutionException("Failed too many times.", query)
+
+    try:
+        context.setQuery(query.load(limit))
+        result = context.query().convert()
+        return result if result else []
+
+    except HTTPError:
+        time.sleep(delay_in_seconds())
+        return execute(query, limit, context, tries - 1)
+
+    except SPARQLExceptions.SPARQLWrapperException as err:
+        raise QueryExecutionException(err.msg, query) from err
+
+    except Exception as err:
+        raise QueryExecutionException(
+            f"{type(err).__name__} - {str(err)}", query
+        ) from err
+
+
+def check_sparql_file(query_file: str) -> Path:
+    """Check meta information of SPARQL query file.
+
+    Args:
+        query_file (str): the file to validate.
+
+    Returns:
+        Path: the validated file.
```

Comment:
I'd change this to `fpath (str): ...`. To me `file_path` is also a bit more in line with the variable/function names. I can make these edits though :)

Reply:
Done.